### DISCUSSION TITLE
GitHub Secrets sometimes empty

### DISCUSSION THREAD

**Discussion by AllexVeldman** [community/community#27026] [Actions]
<p>Recently I started noticing automatically created PR’s (like created by Dependabot) failing due to empty Secrets.</p>
<p>When re-running the jobs all Secrets work again.</p>
<p>The GITHUB_TOKEN does not seem to be affected.</p>
<p><img src="https://user-images.githubusercontent.com/59562474/181101370-dc23e2d6-c90a-4f20-9444-46fecabda1a1.png" alt="image" data-base62-sha1="AriP3YZeOoKRuRAtf9Hg5NJTBlk" width="189" height="62"></p>

**Comment by airtower-luna** [comment, marked as answer]
<p>That change was intentional, because of a potential security problem:</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.blog/wp-content/uploads/2019/01/cropped-github-favicon-512.png?fit=32%2C32" class="site-icon" width="32" height="32">
      <a href="https://github.blog/changelog/2021-02-19-github-actions-workflows-triggered-by-dependabot-prs-will-run-with-read-only-permissions/" target="_blank" rel="noopener">The GitHub Blog</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://github.blog/wp-content/uploads/2019/03/blog-card.png?fit=5001%2C2626" class="thumbnail" width="690" height="362"></div>

<h3><a href="https://github.blog/changelog/2021-02-19-github-actions-workflows-triggered-by-dependabot-prs-will-run-with-read-only-permissions/" target="_blank" rel="noopener">GitHub Actions: Workflows triggered by Dependabot PRs will run with read-only...</a></h3>

<p>GitHub Actions: Workflows triggered by Dependabot PRs will run with read-only permissions</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by AllexVeldman** [comment]
<p>Completely missed that, thank you.</p>

**Comment by fearphage** [comment]
<aside class="quote no-group" data-username="airtower-luna" data-post="2" data-topic="172812">
<div class="title">
<div class="quote-controls"></div>
 airtower-luna:</div>
<blockquote>
<p>That change was intentional,</p>
</blockquote>
</aside>
<p>What’s the fix for this? How do I get back to Dependabot working as expected in the most secure fashion?</p>

**Comment by airtower-luna** [comment]
<p>Depends on the details of your workflow, but the changelog entry links to this article which discusses dangers and possibilities:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://securitylab.github.com/research/github-actions-preventing-pwn-requests/">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/5814919/181101374-79f00bcc-3a9a-4b8d-b67d-49bbd9c1f52b.png" class="site-icon" width="72" height="72">

      <a href="https://securitylab.github.com/research/github-actions-preventing-pwn-requests/" target="_blank" rel="noopener" title="12:00AM - 15 December 2020">GitHub Security Lab – 15 Dec 20</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://user-images.githubusercontent.com/5814919/181101375-c152436a-78dd-416b-a08e-e7be04bda3e1.png" class="thumbnail" width="690" height="362" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/e/c/ec60f25ad72688253445273a7394448fe64ea35a_2_690x362.png, https://aws1.discourse-cdn.com/github/optimized/3X/e/c/ec60f25ad72688253445273a7394448fe64ea35a_2_1035x543.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/e/c/ec60f25ad72688253445273a7394448fe64ea35a.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/e/c/ec60f25ad72688253445273a7394448fe64ea35a_2_10x10.png"></div>

<h3><a href="https://securitylab.github.com/research/github-actions-preventing-pwn-requests/" target="_blank" rel="noopener">Keeping your GitHub Actions and workflows secure: Preventing pwn requests</a></h3>

  <p>In this article, we’ll discuss some common security malpractices for GitHub Actions and workflows, and how to best avoid them. Our examples are based on real-world GitHub workflow implementation vulnerabilities the GitHub Security Lab has reported to...</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
